packages/react-native/src/evm/providers/full.tsx

```diff
@@ -0,0 +1,356 @@
+import type { QueryClient } from '@tanstack/react-query';
+import {
+  ThirdwebSDKProvider,
+  ThirdwebSDKProviderProps,
+  ThirdwebConfigProvider,
+  ThirdwebAuthConfig,
+  SupportedChain,
+  defaultSupportedChains,
+  Chain,
+} from '@thirdweb-dev/react-core';
+import {
+  DEFAULT_RPC_URLS,
+  SDKOptions,
+  getProviderForNetwork,
+  SDKOptionsOutput,
+} from '@thirdweb-dev/sdk';
+import type { ThirdwebStorage } from '@thirdweb-dev/storage';
+import React, { useMemo } from 'react';
+import {
+  Chain as WagmiChain,
+  WagmiConfig,
+  useProvider,
+  Connector,
+  createClient,
+} from 'wagmi';
+import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';
+import { getDefaultProvider } from 'ethers';
+import invariant from 'tiny-invariant';
+import { getWagmiChain } from '../utils/chains';
+import { TW_WC_PROJECT_ID, WC_RELAY_URL } from '../constants/walletConnect';
+
+/**
+ * @internal
+ */
+export type WalletConnectConnectorType =
+  | WalletConnectConnector
+  | 'walletConnect'
+  | { name: 'walletConnect'; options: WalletConnectConnector['options'] };
+
+/**
+ * @internal
+ */
+export type WalletConnector = WalletConnectConnectorType;
+
+/**
+ * @internal
+ */
+export type ChainRpc<TSupportedChain extends SupportedChain> = Record<
+  TSupportedChain extends Chain ? TSupportedChain['id'] : TSupportedChain,
```

Comment:
i think this changed slightly with the anyEVM branch - we should probably consolidate since that's coming out this week. cc @jnsdls

Reply:
Ok, I'll wait to merge